documentation/docs/guides/isolated-development-environments.md

```diff
@@ -0,0 +1,223 @@
+"I want to experiment with adding a new feature, but I want to do it in an isolated environment so I don't affect my main codebase."
+```
+
+Goose will automatically:
+1. Create a new git branch for your work
+2. Set up a containerized environment
+3. Ensure all changes are isolated from your host system
+
+### Working with Experiments
+
+```
+"Let me try a completely different approach to this algorithm. Can you set up an isolated environment where I can experiment?"
+```
+
+### Learning New Technologies
+
+```
+"I want to try out this new framework, but I don't want to install all its dependencies on my main system."
+```
+
+## Benefits
+
+- **Safety**: Experiment without fear of breaking your main development environment
+- **Reproducibility**: Consistent environments across different machines and team members
+- **Isolation**: Multiple projects can run simultaneously without conflicts
+- **Easy cleanup**: Remove containers and branches when done
+- **Version control**: All changes are tracked in isolated git branches
+- **Rollback capability**: Easily discard failed experiments
+
+## Common Workflows
+
+### Feature Development
+
+1. Start a conversation with Goose about a new feature
+2. Request isolated development environment
+3. Goose creates branch and container
+4. Develop and test the feature
+5. If successful, merge the branch; if not, discard it
+
+### Dependency Exploration
+
+1. Ask Goose to explore a new library or tool
+2. Work in isolated container with the dependency
+3. Test compatibility and functionality
+4. Decide whether to integrate into main project
+
+### Refactoring
+
+1. Request isolated environment for major refactoring
+2. Make changes in safety of container and branch
+3. Test thoroughly before merging
+4. Easy rollback if issues arise
+
+## Troubleshooting
+
+### Common Issues
+
+**Docker not running:**
+- Ensure Docker Desktop is installed and running
+- Check Docker daemon status: `docker info`
+
+**Permission issues:**
+- Ensure your user has permission to run Docker commands
+- On Linux, add user to docker group: `sudo usermod -aG docker $USER`
+
+**Extension not working:**
+- Verify Container Use is installed: `cu --version`
+- Check your Goose configuration file syntax
+- Restart Goose after adding the extension
+
+**Git issues:**
+- Ensure Git is properly configured with user name and email
+- Check that you're in a Git repository when starting isolated work
+
+### Getting Help
+
+If you encounter issues:
+
+1. Check the **[Container Use GitHub repository](https://github.com/dagger/container-use)** for documentation
+2. Verify all prerequisites are installed and working
+3. Check Goose logs for error messages
+4. Join our [Discord community](https://discord.gg/block-opensource) for support
+
+## Advanced Configuration
+
+### Custom Container Images
+
+You can configure container-use to use specific base images for different types of projects by setting environment variables in your extension configuration.
```

Comment:
For now, the agent can change its base image on its own (e.g. it defaults to an `ubuntu`, but based on the task at hand it can decide to switch to `python:3.13-slim` or whatnot).

We have a mechanism for the user to set their own base image but it's not documented yet because the interface is not stable